base/commands/update_check.go

```diff
@@ -0,0 +1,110 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc/paths"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/store"
+	"github.com/hazelcast/hazelcast-commandline-client/internal"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+)
+
+const skipUpdateCheck = "CLC_SKIP_UPDATE_CHECK"
+
+const newVersionWarning = `A newer version of CLC is available.
+Visit the following link for release notes and to download:
+https://github.com/hazelcast/hazelcast-commandline-client/releases/%s
+`
+
+const (
+	updateCheckKey   = "update.nextCheckTime"
+	updateVersionKey = "update.latestVersion"
+	checkInterval    = time.Hour * 24 * 7
+)
+
+func maybePrintNewerVersion(ctx context.Context, ec plug.ExecContext) error {
```

Comment:
```suggestion
func maybePrintNewVersionNotification(ctx context.Context, ec plug.ExecContext) error {
```
Could you add a test for this function?

Reply:
Added a test case that does not make a call to the Github API: https://github.com/hazelcast/hazelcast-commandline-client/pull/294/commits/23466d7ad9ddc302f3576b5e39f1690f0be76ef5.